src/tyr/actor.cc

```diff
@@ -18,6 +18,10 @@ struct actor_t::pimpl_t {
       : reader(new baldr::GraphReader(config.get_child("mjolnir"))), loki_worker(config, reader),
         thor_worker(config, reader), odin_worker(config) {
   }
+  pimpl_t(const boost::property_tree::ptree& config, baldr::GraphReader& graph_reader)
```

Comment:
for gurka, and other users, we allow you to pass in a graphreader instead of creating one. this is handy when you already have a reader around and want to re-use it. especially if you want to return pointers into data structures you want them to persist after the call. this isnt the case in gurka but can be the case in other uses of the library